client/powerline.c

```diff
@@ -90,8 +91,18 @@ int main(int argc, char *argv[]) {
 	memset(&server, 0, sizeof(struct sockaddr_un));
 	server.sun_family = AF_UNIX;
 	strncpy(server.sun_path A, address, strlen(address));
+#ifdef __linux__
+	/* We use a socket in the abstract namespace on Linux, and because SUN_LEN uses strlen, it does not work for these. The struct size also fails, so compute manually. */
+	socklen = sizeof(server.sun_family) + strlen(address) A;
+#elif defined SUN_LEN
```

Comment:
All three variants give the same result:

echo -E '#include <stddef.h>'$'\n''typedef struct {char b; char a[100];} A; main() { A *v; printf("oo:     %i\nosx:    %i\nGentoo: %i", offsetof(A, a), (int) (sizeof(*v)-sizeof(v->a)), (int) (((A*)0)->a)); }' | tcc -run -

(same for `clang -xc - && ./a.out` and `gcc`). But `offsetof` is clearly cleaner.